rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/azure/AzureBakeHandler.groovy

```diff
@@ -98,21 +102,28 @@ public class AzureBakeHandler extends CloudProviderBakeHandler{
     ]
 
     if (bakeRequest.build_number && bakeRequest.base_name) {
-      parameterMap.azure_image_name = "$bakeRequest.build_number-$bakeRequest.base_name"
+      parameterMap.azure_managed_image_name = "$bakeRequest.build_number-$bakeRequest.base_name"
     } else if (imageName) {
-      parameterMap.azure_image_name = imageName
+      parameterMap.azure_managed_image_name = imageName
     } else {
-      parameterMap.azure_image_name = Clock.systemUTC.millis().toString()
+      parameterMap.azure_managed_image_name = Clock.systemUTC.millis().toString()
     }
 
-    // Ensure 'azure_image_name' conforms to CaptureNamePrefix regex in packer.
+    // Ensure 'azure_managed_image_name' conforms to CaptureNamePrefix regex in packer.
     // https://github.com/mitchellh/packer/blob/master/builder/azure/arm/config.go#L45
-    def azureImageName = parameterMap.azure_image_name
+    def azureImageName = parameterMap.azure_managed_image_name
     azureImageName = azureImageName.replaceAll(/[^A-Za-z0-9_\-\.]/, "")
-    azureImageName = azureImageName.length() <= 23 ? azureImageName : azureImageName.substring(0, 23)
     azureImageName = azureImageName.replaceAll(/[\-\.]+$/, "")
-
-    parameterMap.azure_image_name = azureImageName
+    // Cut the 75 characters for image name due to Azure Managed Disk limitation
+    // The imageName is made up by 4 parts: PACKAGE_NAME-PACKAGE_TYPE-TIMESTAMP-OS
+    // A potential issue is that if the package name is too long(let's say 80), then if we cut the name by 75
+    // the final image name will only contains part of the package name. So it will be the same for every build
+    // A better way to fix is cutting the package name if it is too long and always keep the "PACKAGE_TYPE-TIMESTAMP-OS" part
+    // But since the imageName here is already combination of 4 parts, not easy to figure out what the exact package name is
+    azureImageName = azureImageName.length() <= 75 ? azureImageName : azureImageName.substring(0, 75)
```

Comment:
This being Groovy, you can replace this line with:
`azureImageName = azureImageName.take(75)`

http://docs.groovy-lang.org/docs/groovy-2.3.2/html/api/org/codehaus/groovy/runtime/StringGroovyMethods.html#take(java.lang.CharSequence,%20int)